### PR TITLE
Pyenv installation script

### PR DESCRIPTION
**Pull request by adobrzhansky**
Basic script to install pyenv on a fresh env

### CODE REVIEW

**Review by rafie — APPROVED** [review]

File: bin/getpyenv [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+profile_d=`get_profile_d`
+if [[ -f $profile_d/pyenv.sh && $FORCE != 1 ]]; then
+	echo "# Already installed."
+else
+	if [[ $THIN != 1 ]]; then
+		if is_command apt-get; then
+			xinstall git make build-essential libssl-dev zlib1g-dev \
+				libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
+				libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
```
Comment (chayim) [inline bin/getpyenv:49]: liblzma-dev
libxmlsec1-dev
llvm

are not python dependencies

gdbm is missing

On older versions of Ubuntu you need tcl-dev not just tk dev

File: bin/getpyenv [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+				xz-dev readline-dev sqlite-dev tk-dev linux-headers 
+		elif is_command pacman; then
+			xinstall base-devel openssl zlib xz tk
+		elif is_command brew; then
+			xinstall openssl readline sqlite3 xz zlib tcl-tk
+		fi
+	fi
+
+	if [[ ! -r $HOME/.pyenv ]]; then
+		runn git clone https://github.com/pyenv/pyenv.git ~/.pyenv
```
Comment (chayim) [inline bin/getpyenv:70]: Support specific pyenv tags. pyenv master is frequently unstable.

File: bin/getpyenv [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+		fi
+		END
+	add_to_profile_d $tmp_profiled/pyenv.sh
+	rm -fr $tmp_profiled/pyenv.sh
+	echo "# Done."
+fi
+
+if [[ -n $PYTHON ]]; then
+    echo "# Installing python $PYTHON ..."
+	$OP . $profile_d/pyenv.sh
```
Comment (chayim) [inline bin/getpyenv:89]: Spacing doesn't align

File: bin/getpyenv [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+fi
+
+if [[ -n $VENV ]]; then
+	if [[ -e $VENV ]]; then
+		eprint "Directory $VENV exists, virtualenv not installed."
+		exit 1
+	fi
+    echo "# Installing virtualenv ..."
+	$OP export PYENV_VERSION="$PYTHON"
+	runn python3 -m pip install virtualenv
```
Comment (chayim) [inline bin/getpyenv:100]: Python comes with virtualenv. It's the venv library, a built in as of [python 3.3](https://docs.python.org/3/library/venv.html)
```python -m venv <name of venv```

Comment (rafie) [reply bin/getpyenv:100]: @chayim I was under the impression `virtualenv` has some mojo over `venv` and it's the virtualenv package of choice. Is this not true anymore?

File: bin/getpyenv (hunk 2 above)
Comment (rafie) [reply bin/getpyenv:70]: @chayim This comes from their installation instructions in gitub, trying to mimic the manual experience.
Do you think pinning a specific version and updating it from time to time is a better tactic?
For instance, we can go with this one until the 1st time it breaks and then switch to pinning ("always optimistic" :-) ).

File: bin/getpyenv (hunk 1 above)
Comment (rafie) [reply bin/getpyenv:49]: @chayim This is coming from their installation instructions (https://github.com/pyenv/pyenv/wiki#suggested-build-environment).
For comparison, your cpython build in Gears does include `liblzma-dev` and `libgdbm-dev` (which is missing here).
llvm looks indeed an overkill.
`libxmlsec1-dev` is missing from Gears but if they insist I'll yield.

File: bin/getpyenv (hunk 4 above)
Comment (rafie) [reply bin/getpyenv:100]: Like, we use `virtualenv` everywhere. Should we switch to `venv`?

**Pull request merged**